### ISSUE TITLE
Fix yaml.safe_load() deprecation warnings

### ISSUE THREAD

**Issue by janosh** [materialsproject/pymatgen#2359]
My `pytest` and `fireworks`  logs are full of `yaml.safe_load()` deprecation warnings originating in `pymatgen` and `monty`.

```py
site-packages/monty/serialization.py:85: PendingDeprecationWarning: 
  load will be removed, use
  
    yaml=YAML(typ='unsafe', pure=True)
    yaml.load(...)
  
  instead
    return yaml.load(fp, *args, **kwargs)

site-packages/pymatgen/analysis/local_env.py:51: PendingDeprecationWarning: 
  safe_load will be removed, use
  
    yaml=YAML(typ='safe', pure=True)
    yaml.load(...)
  
  instead
    default_op_params = yaml.safe_load(f)

site-packages/pymatgen/analysis/local_env.py:54: PendingDeprecationWarning: 
  safe_load will be removed, use
  
    yaml=YAML(typ='safe', pure=True)
    yaml.load(...)
  
  instead
    cn_opt_params = yaml.safe_load(f)
```

Would be nice to fix those to have readable logs again. Should be doable by turning this

https://github.com/materialsproject/pymatgen/blob/905fa8592423f51784b9e83e02c53bb0c6ea0aa3/pymatgen/core/__init__.py#L11-L17

into this

```py
try:
    from ruamel.yaml import YAML

    yaml = YAML(typ="safe", pure=True)
except ImportError:
    try:
        import ruamel_yaml as yaml  # type: ignore  # noqa
    except ImportError:
        import yaml  # type: ignore # noqa
```

and then replace all `yaml.safe_load()` with `yaml.load()`. What do you say?

**Comment by janosh** [comment]
Btw, `pyyaml` [also exports a `load()` function](https://pyyaml.org/wiki/PyYAMLDocumentation) so it wouldn't throw an error to use `pyyaml.load()` for people who don't have `ruamel.yaml` but it would no longer be "safe". Not sure if that's an issue? I guess we could monkey-patch it:

```py
try:
    from ruamel.yaml import YAML

    yaml = YAML(typ="safe", pure=True)
except ImportError:
    try:
        import ruamel_yaml as yaml  # type: ignore  # noqa
    except ImportError:
        import yaml  # type: ignore # noqa

        yaml.load = yaml.safe_load
```

**Comment by janosh** [comment]
Closed in d7cfd8a.
